lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -1107,14 +1107,9 @@ WasmBytecodeGenerator::EmitReturnExpr(EmitInfo *lastStmtExprInfo)
         }
 
         m_writer.Conv(retOp, 0, retExprInfo.location);
-        if (!lastStmtExprInfo)
-        {
-            ReleaseLocation(&retExprInfo);
-        }
     }
     m_writer.AsmBr(m_funcInfo->GetExitLabel());
-
-    return EmitInfo();
+    return retExprInfo;
```

Comment:
It seems that sexpr-wasm also passes the original example. Moreover, labels.wast's misc2 also exhibits internal return, which surprisingly type checks.